### PR TITLE
feat: ViZDoom PPO training

### PR DESCRIPTION
**Pull request by P-H-B-D**
Simpler training and inference of ViZDoom PPO agent.

### CODE REVIEW

File: ViZDoomPPO/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,127 @@
+absl-py==2.1.0
+aiohappyeyeballs==2.4.0
+aiohttp==3.10.5
```
Comment (arnaudstiegler) [inline ViZDoomPPO/requirements.txt:3]: Any chance we can align that with the repo requirements? AFAIK, there's no actual need for a vizdoom/stablebaseline specific environment

File: ViZDoomPPO/common/models.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import gym
+import torch as th
+from stable_baselines3.common.torch_layers import BaseFeaturesExtractor
+from torch import nn
+
+from common.monitoring import register_hooks
+
+
+class CustomCNN(BaseFeaturesExtractor):
```
Comment (arnaudstiegler) [inline ViZDoomPPO/common/models.py:7]: This doesn't seem to be used anywhere

File: ViZDoomPPO/loadrun.py [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+        self._print_state()
+
+        state = super().reset()
+
+        self.last_health = 100
+        self.last_x, self.last_y = self._get_player_pos()
+        self.last_armor = self.last_frags = self.total_rew = self.deaths = 0
+
+        # Damage count  is not cleared when starting a new episode: https://github.com/mwydmuch/ViZDoom/issues/399
+        # self.last_damage_dealt = 0
```
Comment (arnaudstiegler) [inline ViZDoomPPO/loadrun.py]: Is it intended to have that line commented out?

File: ViZDoomPPO/loadrun.py [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+    dataset = Dataset.from_list(entries, features=features)
+
+    # Save to Parquet
+    dataset.to_parquet(parquet_path)
+
+    print(f"Data saved to {parquet_path}")
+    env.close()
+
+
+MODEL_PATH = "/Users/pbowmandavis/Downloads/gameNgen-repro/stable_baselines_playground/rl-doom/standalone_examples/logs/models/dm_longrun_test/final_model.zip"
```
Comment (arnaudstiegler) [inline ViZDoomPPO/loadrun.py]: use relative path here

**Review by arnaudstiegler — COMMENTED** [review]
Looks good, few asks on my end:
- can you remove all files / classes that are not used for training / collecting data
- can you add some documentation in the repo README to document the entrypoints (what should I run to trian the model, the viz etc...)
- [Can be a separate PR] Add a script that takes the output of running the agent in the env, and generates the HF hub dataset

File: ViZDoomPPO/requirements.txt (hunk 1 above)
Comment (P-H-B-D) [reply ViZDoomPPO/requirements.txt:3]: let's handle in next PR if ok

File: ViZDoomPPO/load_model_generate_dataset.py [hunk 5]
```diff
@@ -422,73 +421,149 @@ def make_gif(agent, file_path, num_episodes=1):
...
 
 if __name__ == "__main__":
+    args = parse_arguments()
+
     scenario = "deathmatch_simple"
 
     env_args = {
         "scenario": scenario,
-        "frame_skip": 2,
+        "frame_skip": 1,
```
Comment (arnaudstiegler) [inline ViZDoomPPO/load_model_generate_dataset.py:582]: Shouldn't that be 4 instead? Should probably be a constant stored in a config file

File: ViZDoomPPO/load_model_generate_dataset.py [hunk 6]
```diff
@@ -1,19 +1,3 @@
...
-#  |  $$ / $$/| $$|____ /$$/| $$  | $$ /$$__  $$ /$$__  $$| $$_  $$_  $$      | $$$$$$$/| $$$$$$$/| $$  | $$   #
-#   \  $$ $$/ | $$   /$$$$/ | $$  | $$| $$  \ $$| $$  \ $$| $$ \ $$ \ $$      | $$____/ | $$____/ | $$  | $$   #
-#    \  $$$/  | $$  /$$__/  | $$  | $$| $$  | $$| $$  | $$| $$ | $$ | $$      | $$      | $$      | $$  | $$   #
-#     \  $/   | $$ /$$$$$$$$| $$$$$$$/|  $$$$$$/|  $$$$$$/| $$ | $$ | $$      | $$      | $$      |  $$$$$$/   #
-#      \_/    |__/|________/|_______/  \______/  \______/ |__/ |__/ |__/      |__/      |__/       \______/    #
-#                                                                                                              #
-# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #                                                                                                
-                                                                                                         
-# FORK OF LEANDRO KIELIGER'S DOOM PPO TUTORIAL: https://lkieliger.medium.com/deep-reinforcement-learning-in-practice-by-playing-doom-part-1-getting-started-618c99075c77
```
Comment (arnaudstiegler) [inline ViZDoomPPO/load_model_generate_dataset.py]: If this was taken from someone else, I recommend keeping the reference

File: README.md [hunk 7]
```diff
@@ -3,9 +3,17 @@
...
 
-## Run training
+## ViZDoom PPO Agent Operations
+Train model: `python ./ViZDoomPPO/train_ppo_parallel.py`
+
+Load trained model and upload dataset: `python ./ViZDoomPPO/load_model_generate_dataset.py --output parquet --episodes <n> --upload --hf_token <hf token> --hf_repo <repo name>`
+
+Example of dataset loading: `python ./ViZDoomPPO/explore_dataset.py`
+
+## Run vision model training
```
Comment (arnaudstiegler) [inline README.md]: nit: `train diffusion model`

File: ViZDoomPPO/load_model_generate_dataset.py [hunk 8]
```diff
@@ -422,73 +421,149 @@ def make_gif(agent, file_path, num_episodes=1):
...
+        writer.close()
+
+    print(f"Data saved to {output_parquet}")
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Generate GIF or Parquet file from pretrained PPO agent")
+    parser.add_argument("--output", choices=["gif", "parquet"], required=True, help="Output format")
+    parser.add_argument("--episodes", type=int, default=1, help="Number of episodes to run")
+    parser.add_argument("--upload", action="store_true", help="Upload the output to Hugging Face Hub")
+    parser.add_argument("--hf_token", help="Hugging Face API token")
```
Comment (arnaudstiegler) [inline ViZDoomPPO/load_model_generate_dataset.py]: You can usually use the HF_TOKEN environement variable for this, so I'd make this an optional argument

**Review by arnaudstiegler — APPROVED** [review]
Just a few nits, lgtm otherwise

File: ViZDoomPPO/load_model_generate_dataset.py (hunk 5 above)
Comment (P-H-B-D) [reply ViZDoomPPO/load_model_generate_dataset.py:582]: For training, frame skip 4 allows better convergence, but in evaluation I think you always want frame skip 1.

**Pull request merged**